scripts/zkai_d128_snark_ivc_statement_receipt_gate.py

```
@@ -0,0 +1,740 @@
+    expect_equal(payload["result"], RESULT, "result")
+    expect_equal(payload["source_contract"], source_contract(), "source contract", layer="source_contract")
+    expect_equal(payload["non_claims"], NON_CLAIMS, "non claims")
+    expect_equal(payload["validation_commands"], VALIDATION_COMMANDS, "validation commands")
+    require_object(payload["repro"], "repro")
+    inventory = require_list(payload["mutation_inventory"], "mutation inventory")
+    expect_equal(tuple((item.get("mutation"), item.get("surface")) for item in inventory), EXPECTED_MUTATION_INVENTORY, "mutation inventory")
+    cases = require_list(payload["cases"], "cases")
+    expect_equal(len(cases), len(EXPECTED_MUTATION_INVENTORY), "case count")
+    expect_equal(payload["case_count"], len(cases), "case_count")
+    if not payload["all_mutations_rejected"] or not all(case.get("rejected") is True for case in cases):
+        raise D128SnarkReceiptError("not all SNARK receipt mutations rejected", layer="mutation_suite")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)